llvm/lib/MC/MCC/AssemblerC.cpp

```diff
@@ -301,11 +301,6 @@ LLVMBool LLVMDisassembleEraVM(LLVMTargetMachineRef T,
       break;
   }
 
-#ifndef NDEBUG
-  if (ConstantSectionStart != std::numeric_limits<uint64_t>::max())
-    assert(PC == ConstantSectionStart * WordSize);
-#endif
-
```

Comment:
I removed this assertion, as it's not valid in general case. The condition may be broken when passing trash binary to disassembler.

Reply:
Perhaps, we should extract it to a separate fixup commit as irrelevent to this patch.